File: 7 Reverse Integer/7reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1 if x < 0 else 1
        int_M = 2147483648
        int_m = -2147483648
        rev = 0
        x *= sign
        c = len(str(x))

        while x:
            digit = x % 10
            x = x //10
            rev = rev*10 + digit
        
        rev *= sign
        
        if c == 10:
            if rev + 1 > int_M:
                return 0
            if rev < int_m:
                return 0
        return rev
            

File: 7 Reverse Integer/7reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        vector<int> rev;
        int a = 0, m = 1; 

        while (x != 0) {
            rev.push_back(x % 10);
            x /= 10;
        }

        for (int i = 0; i < rev.size(); i++){
            if (a > INT_MAX / 10 || (a == INT_MAX / 10 && rev[i] > 7)){return 0;}
            if (a < INT_MIN / 10 || (a == INT_MIN / 10 && rev[i] < -8)){return 0;}
            a = a * 10 + rev[i];
        }
        return a;
    }
};